### ISSUE TITLE
Form checkValidity() sometimes returns true when form fields are invalid

### ISSUE THREAD

**Issue by kevinkace** [MithrilJS/mithril.js#2256]
When using [`m.withAttr()` to maintain an `input` `value`](https://github.com/MithrilJS/mithril.js/blob/next/docs/withAttr.md#description), the form's `checkValidity()` method sometimes returns `true` when form fields are not valid. This happens on the second and subsequent submissions.

## Expected Behavior

`checkValidity()` should always return `false` if any fields within the `form` aren't valid, eg: https://codepen.io/kevinkace/pen/rqKroe?editors=1011

## Current Behavior

`checkValidity()` returns `true` when a field contains an invalid `value`, on the second and subsequent submits. eg: https://codepen.io/kevinkace/pen/XxYBoK?editors=0011

## Steps to Reproduce (for bugs)

https://codepen.io/kevinkace/pen/XxYBoK?editors=0011

1. use `m.withAttr()` to set the `value` of an `input` with a `minlength`, eg:
```js
m("input", {
  minlength : 4,
  value : vnode.state.value,
  oninput : m.withAttr("value", function(value) {
    vnode.state.value = value;
  }
})
```
2. add an `onsubmit` handler to the `form` element, eg:
```js
m("form", {
  oncreate : function(formVnode) {
    vnode.state.formDom = formVnode.dom;
  },
  onsubmit : function(e) {
    e.preventDefault();
    vnode.state.formDom.checkValidity();
  },
  // form elements...
)
```
3. add a `button` to the form, eg `m("button", "submit")`
4. add 3 characters to the input
5. click the `button`, `checkValidity()` should return `false`
6. click the `button`, `checkValidity()` now returns `true`

## Context

I'm trying to used HTML form validation for form submition.

## Your Environment

* Version used: 1.1.6
* Browser Name and version: Chrome 70, Firefox Quantum 62
* Operating System and version (desktop or mobile): Windows 10
* Link to your project: https://github.com/kevinkace/lyrite/tree/firebase (code of issue isn't yet submitted)

**Comment by isiahmeadows** [comment]
So here's what I see: when you click the `button[type=submit]`, it no longer sees the input as the active element, and thus doesn't think to check that the values are equal. If we were to remove the [`$doc.activeElement === vnode.dom` check here](https://github.com/MithrilJS/mithril.js/blob/next/render/render.js#L686), that'd likely solve your problem.

**Comment by isiahmeadows** [comment]
Personally, I'd like to double-check whether `document.activeElement` really needs checked [here](https://github.com/MithrilJS/mithril.js/blob/next/render/render.js#L749-L751) as well.

The check [here](https://github.com/MithrilJS/mithril.js/blob/next/render/render.js#L862) and [here](https://github.com/MithrilJS/mithril.js/blob/next/render/render.js#L871-L872) is required to work around [this IE/old Edge bug](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/9548332/) and [an IE9 bug referenced here](https://allyjs.io/tutorials/mutating-active-element.html), and probably should have a comment added for it at least.

------

But in summary, it's just a case where we're unnecessarily checking `document.activeElement`, and I suspect there might be others, too.

**Comment by fuzetsu** [comment]
@isiahmeadows I agree with your assessment. The `document.activeElement` in `setAttr` seems unnecessary.

[removing document.activeElement seems to work](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIAW94AcyoHoCAnAQwHcBzAS3gDpYB7AWwIFlaNjqoApAZQLNO3KHQBWYAgDsIAD3hCRPCWBAAaEADMeENagDaoaaWYQ0IOlmZQNIJtMSOLIAL7rjp8+lV2HT+AsYeAACLUZiZnUQ6mkcAFd4AG4AHWk05jpmRnjHAAokRlh4s0c6ACNGJABPaOA0kJDGaVhiCFJEPIA3aSqIAEoQ+ulGxvDIkIBeEJ6+ukLmVJGQ9waZ6ghybt6kAaG1xrb4eOIR5jyU7QjmS7qD0cbertIoaiQOiBDUEPhieIh1PcHs0wPFysJ4Hk9sMHrCQhA6Dg2l0II4ACIQLSkeJQSH9JZwh5A2GwDBwADWeXxxMarmJq2WD3Ol1iCXgt32jNh8GqOE+30uiAUtxpMxe-y+Mx2CLA8A+dGeUH+gK5TNiMGklHgGElABYVYSQm0AI7xahtJCS37K0XNVmJSWZcicACC2GIFxAiv+HO64r2kwAfJzDY1Zrs6LL5d7PtMYwTQ6SKVSE3DXP0DYTmq12p17fAAGrSwYw0P5qYxOKJItzBap2F01UrDPE5kgcqJeDNDnAH68-khS6g8G0S4raJDsEQy79e7U5aN9NLNJaXIIajNEJJ2CUktrBxgRgwOhQRiUPIAAy7jH4GAi8G+ABJgPmFS83rRqnRr7f764L5moajBeMYcGAYCxJQT4vlW9CKh+PJvkqEBgRBmr-oBQEhBe27kgW75IJ+0HjJkuH4a8hE8lS-5rPOjZpHYkAwOuIIWAAjAAzKgACcbgeCAJhmBYDDgX4zQBC4rgALqaK80jkvoRgCV4FgQlwKiSHYJy2OgWC4PgRBkFQtAMCw7DKHwghqaIqgyPIijWRpaiaDyfIWGArTUDggSaDgHRJopBgGAADJo-ogDJBgALRsZoIQAGTxVKtYsFMkzTI+hSwHQpDriiACiMClIEkWhSAgxtBF6i6vqAAc1XBQArAATPqupNa1UkyeA0BwPAG7SPoIBscFABsADsqC6m4UmuEAA)

**Comment by StephanHoyer** [comment]
[still an issue](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA83KAD4AOnQAEUwVDKjuABwjT4WCFJ5IIK6TzpgRmgBQAjbvHjdcUmCXgBKQQHp5E6bPcMkSKQy0sBgAnBgQIYKlrKI0pMjolAFd4V3dJGTkFWHlYAGspcXBEs1wyeELCKW4zSGCAN01YDTyANQZ5JDKATxNHO24AcwH4gakSdshUhXSvLJz8wrBi0vKQfylIAz94MmZK6tqGwLhcto7u3v6hkajgxIgpj1cRD0lBTrq4pABeQtxuRJ8QqiVwfUScSAwBBkXhUdAAdlQACYkSAAL6EehMFjoDAAKyoxAM-HgrBg8DG3GCuEq8SS8AA3JJJLgMP9AfATOy+JVgDNeLBghAGIgTHU6NwdH0+Z4ZCQqbZvlJxZKIBgkDYmZ4MTM6mQIAB3MUSqVSGUyGRC+CJYLSXAmQry6kVM0zC0WiV1dpkJAizSoW73Qhu91VQzLMomCDSkOhqRqpRChp8AAiEHGiSgnMcWrj7tjoaaJ16ubzaILOtlFvthTpyRd5rz8C6KikAcKiAAHqtg1X3V6oPc28qTWqwPA-RgB0GCzJSnQYHQBuphwAWXt5qRCgCOiTIQr8AfgdwgG7zvDrFIDUeTTik31Err7oZVOgw48n080Sogt4wE+CAYIHgKd2nuUtNyLPIS1nKRK03AUhT9ExLxaUcY2fd1L3vOIEmSNDVXVTVYPLZ80UcM93RrEAzGSaw6AbKIW39AoihKMpCjgypFgjVZHBDHMZlI8itUkEhAWhXhjmgjCZAMIwYAwKBBhMAADaxuAAZSwKl4ADAASYBL1A85m3-bgtJ04J4DRVTKM3C1VK-ABZMgwDAEYDKMvCQIHH1uhM+5XPckZbPshypFUqDTm9Tpmy8p1WWis5-ObXpbJmQS6FIyR0UxEBGGYVgMFgdzOGJVRSXQdEAF1iHkOhcjhWgCuxVgViwYIKDZeJ8UJEAbQIdAcHgJQwFQFwXFCA0MGGdRikSWoKr4EqbBcVz1C6qAAClNJcDqtr6lw6Agbt9rKTruvnPrOGbFRWDAQUyCUUk0RqtEgA)

**Comment by kfule** [comment]
It appears to be due to #2257 simply not being merged.
https://github.com/MithrilJS/mithril.js/pull/2257#issuecomment-2599762079
